hyrax.gemspec

```diff
@@ -26,6 +26,12 @@ EOF
   spec.version       = Hyrax::VERSION
   spec.license       = 'Apache2'
 
+  # Be pessimistic, rails does not follow sem-ver conventions and
+  # minor version releases can include breaking changes; see
+  # http://guides.rubyonrails.org/maintenance_policy.html
+  # https://github.com/projecthydra-labs/hyrax/issues/1028
+  spec.add_dependency 'rails', '>= 5.0.0', '< 5.2'
```

Comment:
Looks like the votes are in and the accepted practice is to respond to upgrade failures when they happen, so the gemspec will become `~> 5.0` to allow any `5.x` rails version.  That implies that breaking changes in rails minor versions are to be expected and fixed ad-hoc.